components/BuyButton/index.js

```diff
@@ -0,0 +1,62 @@
+import { mutate } from "swr";
+import { buyProtocol, sellProtocol } from "@/lib/api";
+import styled from "styled-components";
+
+const StyledBuyButton = styled.button`
+  background-color: #000000;
+  border-radius: 5px;
+  border: 0.3px solid var(--secondary-color);
+  height: 2rem;
+  width: 6.25rem;
+  color: var(--secondary-color);
+`;
+
+export default function BuyButton({
+  buyerId,
+  sellerId,
+  starshipId,
+  listingId,
+  preis,
+  buyerCredit,
+  sellerCredit,
+  buyerStarships,
+}) {
+  async function onBuy() {
+    if (buyerCredit < preis) {
+      return alert("You don't have enough credits to buy this starship.");
+    }
+    if (buyerStarships.includes(starshipId)) {
+      return alert("You have this starship already.");
+    }
```

Comment:
Nice, that you check if the buyer already has the starship, but what if the buyer wants to have multiple starships (like two death stars)? 😉

Reply:
We were talking about that now... We are going to change some things on the db and we just decided that the users will be able to have more than one of the same starship... so I will have to change this lines of codes...
Nice that you noticed that and tks for the review.